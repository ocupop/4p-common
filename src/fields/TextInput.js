import React, {
  useState,
  useEffect,
} from 'react'
import Skeleton from 'react-loading-skeleton'
import Label from './label'
import useLoading from '../common/hooks/useLoading'

const TextInput = (props) => {
  const {
    className, hint, type, label, placeholder, required, field, form: { errors, touched },
  } = props
  const status = touched[field.name] && errors[field.name] ? 'is-invalid' : ''

  const loading = useLoading()

  return (
    <div className={`form-group ${className}`}>
      <Label label={label} hint={hint} />
      {loading ? (
        <div className="skeleton">
          <Skeleton height={38} />
        </div>
      ) : (
        <input
          className={`form-control ${status}`}
          {...field}
          placeholder={placeholder}
          type={type}
          required={required}
        />
      )}
      {touched[field.name] && errors[field.name] && (
        <div className="invalid-feedback">{errors[field.name]}</div>
      )}
    </div>
  )
}

export default TextInput
