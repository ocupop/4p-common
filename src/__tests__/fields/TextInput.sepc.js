import React from 'react'
import TextInput from '../../fields/TextInput'

const mockFieldObject = {
  name: 'textInput',
  onChange: () => {

  },
  value: '',
}

const mockFormObject = {
  dirty: false,
  errors: {},
  touched: {},
}

const labelValue = 'im a label!'
const hintValue = 'im the hint!'

describe('TextInput component', () => {
  it('should render an input filed', () => {
    const wrapper = shallow(<TextInput
      className="textInput"
      hint={hintValue}
      label={labelValue}
      type="text"
      placeholder="foobar"
      required
      field={mockFieldObject}
      form={mockFormObject}
    />)
    expect(wrapper.children('input')).toHaveLength(1)
  })

  it('renders the correct input and hint value', () => {
    const wrapper = shallow(<TextInput
      className="textInput"
      hint={hintValue}
      label={labelValue}
      type="text"
      placeholder="foobar"
      required
      field={mockFieldObject}
      form={mockFormObject}
    />)

    expect(wrapper.find('label').prop('hint')).toEqual(hintValue)
    expect(wrapper.find('label').prop('label')).toEqual(labelValue)
  })

  it('simulates a touched trigger', () => {
    const wrapper = shallow(<TextInput
      className="textInput"
      hint={hintValue}
      label={labelValue}
      type="text"
      placeholder="foobar"
      required
      field={mockFieldObject}
      form={mockFormObject}
    />)
    expect(wrapper.find('.is-invalid')).toHaveLength(0)

    wrapper.setProps({
      form: {
        dirty: false,
        errors: { textInput: 'Required' },
        touched: { textInput: true },
      },
    })

    expect(wrapper.find('.is-invalid')).toHaveLength(1)
  })
})
