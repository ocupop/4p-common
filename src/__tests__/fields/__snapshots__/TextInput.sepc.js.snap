// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component Matches the snapshot 1`] = `
<div
  className="form-group textInput"
>
  <label
    className="sr-only"
  >
    <span />
    <div
      className="pt-1 ml-1 d-inline-block"
    >
      <i
        className="ri-question-fill text-mid"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </label>
  <input
    className="form-control "
    name="textInput"
    onChange={[Function]}
    placeholder="foobar"
    required={true}
    type="text"
    value=""
  />
</div>
`;
